### ISSUE TITLE
Bugs

### ISSUE THREAD

**Issue by dekan-aleksandr** [JSB-UCLA/scDED#3]
Hi scDEED developers!

Thank you for your nice approach to UMAPs generation. 
However, it is not possible to launch your program due to multiple bugs. 
Can you fix them in the main branch, please?

Things I found:
1) fir_input_data comes out of nowhere. Also, the following code actually destroys initial input_data. I suggest you write a function that processes data properly and use mclapply to speed up processing drastically. I will take much less time.

      input_data <- subset(input_data, idents = cellTypes[1])
      fir_cells <- Seurat::Cells(fir_input_data)
      if(length(fir_cells) > preserved_cells_num){
        sub_cells <- sample(fir_cells, preserved_cells_num, replace = FALSE)
        input_data <- subset(input_data, cells = sub_cells)
      }
      for (i in cellTypes[-1]) {
        next_input_data <- subset(input_data, idents = i)
        next_cells <- Seurat::Cells(next_input_data)
        if(length(next_cells) > preserved_cells_num){
          sub_cells <- sample(next_cells, preserved_cells_num, replace = FALSE)
          next_input_data <- subset(next_input_data, cells = sub_cells)
        }
        input_data <- merge(input_data, next_input_data)
      }
    }

2) Normalization is repeated 2 times for no reason
  input_data <- Seurat::NormalizeData(input_data, normalization.method = "LogNormalize", scale.factor = 10000)
  input_data <- Seurat::NormalizeData(input_data)
  
 3) I didn't go further, but I suppose using mclapply will be beneficial for everything else as well.

Thank you for the paper though, I really like your idea and approach.

**Comment by clee700** [comment]
Hello, 
We have been in progress of a large package update, which is now updated in the main branch. The tutorial has also been updated. 

The two things you pointed out have been fixed; in this version we leave the data processing (like normalization) up to the user. This is to allow more flexibility, as the previous package would normalize the data through Seurat only, and users may want to use a different pipeline. We also decided to use the package foreach for parallelization instead of mclapply.
